### PR TITLE
Fixed has_calls to assert_has_calls

### LINKED ISSUE #20453: The "has_calls" is used in place of "assert_has_calls" in a few places
### Body

As explained in https://github.com/apache/airflow/pull/20428#issuecomment-998714275 we seem to have  number (not big) of tests that use "has_calls" rather than "assert_has_calls". 

:scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: 

What "has_calls" does is acually calling "has_calls" method on the mock :) . Which make them no-assertion tests:

:scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: 

The list of those tests:

- [x] ./tests/providers/google/common/hooks/test_base_google.py:        mock_check_output.has_calls(
- [x] ./tests/providers/google/common/hooks/test_base_google.py:        mock_check_output.has_calls(
- [x] ./tests/providers/google/cloud/transfers/test_sheets_to_gcs.py:        mock_sheet_hook.return_value.get_values.has_calls(calls)
- [x] ./tests/providers/google/cloud/transfers/test_sheets_to_gcs.py:        mock_upload_data.has_calls(calls)
- [x] ./tests/providers/google/cloud/hooks/test_bigquery.py:        mock_poll_job_complete.has_calls(mock.call(running_job_id), mock.call(running_job_id))
- [x] ./tests/providers/google/cloud/hooks/test_bigquery.py:        mock_schema.has_calls([mock.call(x, "") for x in ["field_1", "field_2"]])
- [x] ./tests/providers/google/cloud/hooks/test_bigquery.py:        assert mock_insert.has_calls(
- [x] ./tests/providers/google/cloud/hooks/test_pubsub.py:        publish_method.has_calls(calls)
- [x] ./tests/providers/google/cloud/hooks/test_cloud_memorystore.py:        mock_get_conn.return_value.get_instance.has_calls(
- [x] ./tests/providers/google/cloud/hooks/test_cloud_memorystore.py:        mock_get_conn.return_value.get_instance.has_calls(
- [x] ./tests/providers/google/cloud/hooks/test_cloud_memorystore.py:        mock_get_conn.return_value.get_instance.has_calls(
- [x] ./tests/providers/google/cloud/hooks/test_dataproc.py:        mock_get_job.has_calls(calls)
- [x] ./tests/providers/google/cloud/hooks/test_dataproc.py:            mock_get_job.has_calls(calls)
- [x] ./tests/providers/google/suite/operators/test_sheets.py:        mock_xcom.has_calls(calls)
- [x] ./tests/providers/http/operators/test_http.py:            mock_info.has_calls(calls)
- [ ] ./tests/providers/airbyte/hooks/test_airbyte.py:        assert mock_get_job.has_calls(calls)
- [ ] ./tests/providers/airbyte/hooks/test_airbyte.py:        assert mock_get_job.has_calls(calls)
- [ ] ./tests/providers/airbyte/hooks/test_airbyte.py:        assert mock_get_job.has_calls(calls)
- [ ] ./tests/providers/airbyte/hooks/test_airbyte.py:        assert mock_get_job.has_calls(calls)
- [ ] ./tests/providers/airbyte/hooks/test_airbyte.py:        assert mock_get_job.has_calls(calls)

We should fix those tests and likelly add pre-commit to ban this.

Thanks to @jobegrabber  for noticing it!

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by harishkrao**
Closes: #20453 

Changed `has_calls` to `assert_has_calls` in the tests mentioned in #20453.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
Let's see if the tests pass :). I have a feeling that some of those might not actually work after the fix :D

**Comment by kaxil** [comment]
Some of the calls might need `any_order=True` --> `mock.assert_has_calls(calls, any_order=True)`

You will have to take a look though - https://docs.python.org/3/library/unittest.mock.html

**Comment by harishkrao** [comment]
> Some of the calls might need `any_order=True` --> `mock.assert_has_calls(calls, any_order=True)`
> 
> You will have to take a look though - https://docs.python.org/3/library/unittest.mock.html

I will make the changes and push it. Thanks @kaxil

**Comment by potiuk** [comment]
Oh yeah. I expected some errors like those in the tests :(

**Comment by harishkrao** [comment]
> Oh yeah. I expected some errors like those in the tests :(

@potiuk , @kaxil I pushed a few changes to mark `any_order=False` in some of the tests. The static checks above failed, any pointers in fixing them would be much appreciated. I ran them in my local breeze environment and they failed there as well. Do I have to configure anything specific for the providers before running pytest?

**Comment by potiuk** [comment]
> > Oh yeah. I expected some errors like those in the tests :(
> 
> @potiuk , @kaxil I pushed a few changes to mark `any_order=False` in some of the tests. The static checks above failed, any pointers in fixing them would be much appreciated. I ran them in my local breeze environment and they failed there as well. Do I have to configure anything specific for the providers before running pytest?

* for statuc check just install `pre-commit` and run it https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#installing-pre-commit-hooks

**Comment by harishkrao** [comment]
@potiuk ran pre-commit on all the changed files, all of them passed. What are the next steps now?

**Comment by potiuk** [comment]
> @potiuk ran pre-commit on all the changed files, all of them passed. What are the next steps now?

The tests need to be fixed - the failing tests now are "unit test", not pre-commits, basically the previous "has_calls" have not checked anything so those tests (or possibly even implementation needs to be fixed).  THis should be individual fix in each case:

a) look at each test why it is failing
b) determine whether the test should be corrected or the implementation
c) fix ...

One-by-one for all failing test cases.

**Comment by harishkrao** [comment]
> > @potiuk ran pre-commit on all the changed files, all of them passed. What are the next steps now?
> 
> The tests need to be fixed - the failing tests now are "unit test", not pre-commits, basically the previous "has_calls" have not checked anything so those tests (or possibly even implementation needs to be fixed). THis should be individual fix in each case:
> 
> a) look at each test why it is failing b) determine whether the test should be corrected or the implementation c) fix ...
> 
> One-by-one for all failing test cases.

@potiuk Thank you for clarifying and listing the steps for resolution. I wasn't sure of what was needed, this is helpful. I will work on this and get back.

File: tests/providers/airbyte/hooks/test_airbyte.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def test_wait_for_job_error(self, mock_get_job):
             self.hook.wait_for_job(job_id=self.job_id, wait_seconds=0)
 
         calls = [mock.call(job_id=self.job_id), mock.call(job_id=self.job_id)]
-        assert mock_get_job.has_calls(calls)
+        assert mock_get_job.assert_has_calls(calls, any_order=False)
```
Comment (uranusjr) [inline tests/providers/airbyte/hooks/test_airbyte.py:90]: Shouldn’t this be

```suggestion
        mock_get_job.assert_has_calls(calls, any_order=False)
```

instead? `assert_has_calls` returns None on success, so asserting its return value is entirely backwards.

**Review by uranusjr — CHANGES_REQUESTED** [review]

File: tests/providers/airbyte/hooks/test_airbyte.py (hunk 1 above)
Comment (harishkrao) [reply tests/providers/airbyte/hooks/test_airbyte.py:90]: Yes, you are correct. I will fix this :) Thank you for pointing me in the right direction. I am new to this provider's code base, so taking time to understand what's needed.

**Comment by uranusjr** [comment]
Did you accidentally merged some unrelated changes to this branch?

**Comment by harishkrao** [comment]
> Did you accidentally merged some unrelated changes to this branch?

I merged Apache: Main into this feature branch, I did not have to do it actually, so I reverted it for now. I am still working on fixing the unit tests one by one, I will comment here once the changes are ready for code review. Apologies if this caused any triggers to notify users @uranusjr

**Comment by potiuk** [comment]
You  should rebase your changes, not merge  - see "How to Rebase PR" https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#id15

**Comment by harishkrao** [comment]
> You should rebase your changes, not merge - see "How to Rebase PR" https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#id15

@potiuk I apologize, it happened erroneously, wasn't intending to do a merge. Will be careful next time. Working on the fix.

**Pull request closed without merging**